src/lib/util/corestr.cpp

```diff
@@ -171,27 +171,41 @@ void strreplacechr(std::string& str, char ch, char newch)
 	}
 }
 
-std::string strtrimspace(std::string& str)
+static std::string internal_strtrimspace(std::string& str, bool right_only)
 {
-	int start = 0;
-	for (auto & elem : str)
-	{
-		if (!isspace(uint8_t(elem)))  break;
-		start++;
-	}
-	int end = str.length();
-	if (end > 0)
+	// identify the start
+	std::string::iterator start = str.begin();
+	if (!right_only)
 	{
-		for (size_t i = str.length() - 1; i > 0; i--)
-		{
-			if (!isspace(uint8_t(str[i]))) break;
-			end--;
-		}
+		start = std::find_if(
+			str.begin(),
+			str.end(),
+			[](char c) { return !isspace(c); });
 	}
-	str = str.substr(start, end-start);
+
+	// identify the end
+	std::string::iterator end = std::find_if(
```

Comment:
This doesn't change the double-scanning of strings containing only spaces at all - it just obfuscates it further.  If you want to do it like this, you need to remap the start to a reverse iterator rather than using `str.rend()` to not scan the same characters twice in that case.